### PR TITLE
ARROW-8721: [CI] Fix R build matrix

### PR DESCRIPTION
**Pull request by kszucs**
If github actions fails to parse he workflow it hides the error by not showing it among the checks.

The workflow has a parsing error: https://github.com/apache/arrow/actions/runs/97349594

### CODE REVIEW

**Review by nealrichardson — CHANGES_REQUESTED** [review]
I'd like to see the build matrix separated like it was before. The two jobs have zero overlap in arguments/parameters, so this indirection has negative value IMO.

**Comment by nealrichardson** [comment]
I do insist, thanks :) I'd rather have a workflow I can understand and debug and shoulder the burden of doing some things twice.

**Comment by kszucs** [comment]
@nealrichardson updated

File: .github/workflows/r.yml [hunk 1]
```diff
@@ -110,6 +88,49 @@ jobs:
         continue-on-error: true
         run: archery docker push ubuntu-r
 
+  rstudio:
+    name: AMD64 CentOS 7 RStudio R 3.6
+    runs-on: ubuntu-latest
+    if: ${{ !contains(github.event.pull_request.title, 'WIP') }}
+    env:
+      R: 3.6
```
Comment (nealrichardson) [inline .github/workflows/r.yml:96]:
```suggestion
```
That env var isn't used in this image

**Review by nealrichardson — APPROVED** [review]
I'm overhauling this file in #7107 so I'll clean up anything else I want over there; this at least should make my merge conflict understandable, so thanks for doing it.

**Review by kszucs — COMMENTED** [review]
+1

**Pull request closed without merging**